### PR TITLE
executor: only show valid columns in `stats_histogram`

### PR DESCRIPTION
**Pull request by eurekaka**
### What problem does this PR solve? 

Fix https://github.com/pingcap/tidb/issues/9479

### What is changed and how it works?

Check if the column is valid or loaded before appending it to result of `show stats_histograms`.

### Check List 

Tests 

 - Unit test

Code changes

N/A

Side effects

N/A

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: executor/show_stats.go [hunk 1]
```diff
@@ -82,6 +82,9 @@ func (e *ShowExec) appendTableForStatsHistograms(dbName, tblName, partitionName
 		return
 	}
 	for _, col := range statsTbl.Columns {
+		if col.IsInvalid(nil, false) {
```
Comment (winoros) [inline executor/show_stats.go]: Why not pass sc here?

Comment (eurekaka) [reply executor/show_stats.go]: Do we need to mark column stats needed in `show stats_histograms`?

Comment (zz-jason) [reply executor/show_stats.go]: how about adding a comment like "pass a nil StatementContext to avoid column stats being marked as needed"?

Comment (eurekaka) [reply executor/show_stats.go]: Comment added, PTAL

**Pull request merged**